--- bfd/Makefile.in	2020-04-20 02:04:13.000000000 +0300
+++ bfd/Makefile-patched.in	2020-12-13 13:56:31.000000000 +0200
@@ -2093,7 +2093,7 @@
 	fi ;\
 	$(SED) -e "s,@bfd_version@,$$bfd_version," \
 	    -e "s,@bfd_version_string@,$$bfd_version_string," \
-	    -e "s,@bfd_version_package@,$$bfd_version_package," \
+	    -e "s|@bfd_version_package@|$$bfd_version_package|" \
 	    -e "s,@report_bugs_to@,$$report_bugs_to," \
 	    < $(srcdir)/version.h > $@; \
 	echo "$${bfd_soversion}" > libtool-soversion
