--- libiberty/simple-object.c	2018-04-13 19:52:10.000000000 +0300
+++ libiberty/simple-object-patched.c	2019-02-04 12:01:37.000000000 +0200
@@ -44,6 +44,10 @@
 #define SEEK_SET 0
 #endif
 
+#ifndef O_BINARY
+#define O_BINARY 0
+#endif
+
 #include "simple-object-common.h"
 
 /* The known object file formats.  */
@@ -326,7 +330,7 @@
       return errmsg;
     }
 
-  outfd = creat (dest, 00777);
+  outfd = open (dest, O_CREAT|O_WRONLY|O_TRUNC|O_BINARY, 00777);
   if (outfd == -1)
     {
       *err = errno;
